phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/LockManager.java

```diff
@@ -21,224 +21,148 @@
+        RowLockImpl lock(long waitDuration) throws IOException{
             synchronized (this) {
-                if (rowLock != null) {
-                    rowLock = new RowLockImpl(this, reentrantLock);
-                    return rowLock;
-                } else {
+                if (!usable) {
                     return null;
                 }
+                count++;
             }
-        }
-
-        void releaseRowLock() {
-            synchronized (this) {
-                if (rowLock != null) {
-                    rowLock.release();
+            boolean success = false;
+            threadName = Thread.currentThread().getName();
+            try {
+                if (!lock.tryLock(waitDuration, TimeUnit.MILLISECONDS)) {
+                    throw new TimeoutIOException("Timed out waiting for lock for row: " + rowKey);
                 }
-            }
-        }
-        
-        void cleanUp() {
-            long c = count.decrementAndGet();
-            if (c <= 0) {
-                synchronized (this) {
-                    if (count.get() <= 0 && rowLock != null){
-                        rowLock = null;
-                        RowLockContext removed = lockedRows.remove(rowKey);
-                        assert removed == this: "we should never remove a different context";
+                success = true;
+            } catch (InterruptedException ie) {
+                LOGGER.warn("Thread interrupted waiting for lock on row: " + rowKey);
+                InterruptedIOException iie = new InterruptedIOException();
+                iie.initCause(ie);
+                Thread.currentThread().interrupt();
+                throw iie;
+            } finally {
+                if (!success) {
+                    synchronized (this) {
+                        count--;
                     }
                 }
             }
-        }
-
-        void setThreadName(String threadName) {
-            this.threadName = threadName;
-        }
-
-        @Override
-        public String toString() {
-            return "RowLockContext{" +
-                    "row=" + rowKey +
-                    ", readWriteLock=" + reentrantLock +
-                    ", count=" + count +
-                    ", threadName=" + threadName +
-                    '}';
-        }
-    }
-
-    /**
-     * Class used to represent a lock on a row.
-     */
-    public static class RowLockImpl implements RowLock {
-        static final RowLockImpl UNINITIALIZED = new RowLockImpl();
-        private final RowLockContext context;
-        private final Lock lock;
-
-        private RowLockImpl() {
-            context = null;
-            lock = null;
-        }
-        
-        RowLockImpl(RowLockContext context, Lock lock) {
-            this.context = context;
-            this.lock = lock;
-        }
-
-        Lock getLock() {
-            return lock;
+            if (success) {
+                return this;
+            }
+            return null;
```

Comment:
Since `InterruptedException` is being re-thrown, `success` here will always be true so we will never return null here. We can directly return `this` instead of checking for `success` value.